charts/alfresco-repository/tests/search_test.yaml

```diff
@@ -23,69 +37,99 @@ tests:
           flavor: solr6
           solr-secret: mysupersecret
           url: https://search.domain.tld/index
-    asserts:
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dindex.subsystem.name=solr6($|\s)
-        template: configmap.yaml
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dsolr\.port=443($|\s)
-        template: configmap.yaml
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dsolr\.base\.url=/index($|\s)
-        template: configmap.yaml
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dsolr\.secureComms=secret($|\s)
-        template: configmap.yaml
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dsolr\.host=search.domain.tld($|\s)
-        template: configmap.yaml
+    template: configmap-search.yaml
+    asserts: !merge-maps
+      - *has_default_search_flavor
+      - equal:
+          path: data.SEARCH_FLAVOR
+          value: solr6
+      - equal:
+          path: data.SEARCH_PORT
+          value: "443"
+      - equal:
+          path: data.SEARCH_HOST
+          value: search.domain.tld
+      - equal:
+          path: data.SOLR_BASE_URL
+          value: /index
+      - equal:
+          path: data.SEARCH_SECURECOMMS
+          value: secret
       - equal:
           path: data.SOLR_SECRET
           value: bXlzdXBlcnNlY3JldA==
         template: secret-search.yaml
+      - matchRegex:
+          path: data.CATALINA_OPTS
+          pattern: |-
+            (^\s*|[^\s]\s+)-Dsolr\.secureComms="\$SEARCH_SECURECOMMS"($|\s)
+        template: configmap.yaml
 
   - it: should render a working Solr6 config based on values (noauth)
     set:
       configuration:
         search:
           << : *solrValues
           solr-secret: null
-    asserts:
-      - matchRegex:
-          path: data.CATALINA_OPTS
-          pattern: |-
-            (^\s*|[^\s]\s+)-Dsolr\.secureComms=none($|\s)
-        template: configmap.yaml
+    asserts: !merge-maps
```

Comment:
what is this?

Reply:
A yaml tag to merge array items (https://github.com/yaml/yaml/issues/48#issuecomment-974998500)

Reply:
and it works in our current setup?

Reply:
Not sure what you mean by that but yes the unit tests pass